packages/gatsby-telemetry/src/error-helpers.js

```diff
@@ -1,11 +1,18 @@
+const { sep } = require(`path`)
+
 // Removes all user paths
-const cleanPaths = str => {
-  // TODO: Recursively remove path sections
-  const localPathRegex = new RegExp(
-    process.cwd().replace(/[-[/{}()*+?.\\^$|]/g, `\\$&`),
-    `g`
-  )
-  return str.replace(localPathRegex, `$PWD`)
+const cleanPaths = (str, separator = sep) => {
+  const stack = process.cwd().split(separator)
+  while (stack.length > 1) {
+    const currentPath = stack.join(separator)
+    const currentRegex = new RegExp(
+      currentPath.replace(/[-[/{}()*+?.\\^$|]/g, `\\$&`),
+      `g`
+    )
+    str = str.replace(currentRegex, `$SNIP`)
```

Comment:
should `$SNIP` be an parameter?

if we need to make it more nice to read when we replace it with an `...`

Reply:
dot's are a bit tricky as those technically valid path segments (tho should be stripped out). with the $SNIP we can see if we somehow end up replacing multiple paths next to eachother